pkgs/applications/misc/waybar/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchFromGitHub, meson, pkgconfig, ninja, 
+  gtkmm3, jsoncpp, libinput, libsigcxx, wayland, wlroots, fmt, 
+  traySupport  ? true,  libdbusmenu-gtk3, 
+  pulseSupport ? false, libpulseaudio,
+  nlSupport    ? true,  libnl,
+  swaySupport  ? false, sway-beta
+}:
+  stdenv.mkDerivation rec {
+    name = "waybar-${version}";
+    version = "0.3.0";
+    src = fetchFromGitHub {
+      owner = "Alexays";
+      repo = "Waybar";
+      rev = version;
+      sha256 = "17dws3a846x6fy2kmxcpa828yn22hf3qi1dlzbcjdl4x2cdmr9dy";
+    };
+
+    nativeBuildInputs = [
+      meson ninja pkgconfig
+    ];
+
+    buildInputs = with stdenv.lib;
+      [ gtkmm3 jsoncpp libinput libsigcxx wayland wlroots fmt ]
+      ++ optional  traySupport  libdbusmenu-gtk3
+      ++ optional  pulseSupport libpulseaudio
+      ++ optional  nlSupport    libnl
+      ++ optional  swaySupport  sway-beta;
+
+    mesonFlags = [ 
+      "-Ddbusmenu-gtk=${ if traySupport then "enabled" else "disabled" }"
+      "-Dpulseaudio=${ if pulseSupport then "enabled" else "disabled" }"
+      "-Dlibnl=${ if nlSupport then "enabled" else "disabled" }"
+      "-Dout=${placeholder "out"}"
+    ];
+
+    meta = with stdenv.lib; {
+      description = "Highly customizable Wayland bar for Sway and Wlroots based compositors.";
```

Comment:
```suggestion
      description = "Highly customizable Wayland bar for Sway and Wlroots based compositors";
```

From the [manual](https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes):

>  Don’t include a period at the end. Don’t include newline characters. Capitalise the first character. For brevity, don’t repeat the name of package — just describe what it does.